docs/data/material/components/buttons/FullWidthButton.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+
+export default function FullWidthButton() {
+  return (
+    <Box
+      sx={{
+        width: '300px',
+        border: '1px solid gray',
+        borderRadius: '5px',
+        padding: '5px',
+      }}
+    >
```

Comment:
To get the inline preview?
```suggestion
   <div>
    <Box
      sx={{
        width: '300px',
        border: '1px solid gray',
        padding: '5px',
      }}
    >
```

Reply:
> I was about to suggest whether we need the 300px width limit there... I think we can make this one span the full demo container width, given it's illustrating the fullWidth prop.

I would:

- move the header from h4 to h3. Even if it's technical under Sizes, IMHO it can also been seen as not, and it would help with anchor linking. 
- rename the header to Full width, since it's what this demo is about
- Increase 300px to some a bit more, up to the limit where it starts to feel poor UX. 
- use "bg": true

![Screenshot 2024-02-04 at 22 46 40](https://github.com/mui/material-ui/assets/3165635/f7d620c3-0058-49e4-87a0-d9b86fd406d6)

Reply:
Not sure about the background, though; it looks cleaner without it (plus all the other demos don't use it).